src/crush/CrushWrapper.cc

```diff
@@ -2186,8 +2186,8 @@ int CrushWrapper::reclassify(
 int CrushWrapper::get_new_bucket_id()
 {
   int id = -1;
-  while (crush->buckets[-1-id] &&
-	 -1-id < crush->max_buckets) {
+  while (-1-id < crush->max_buckets &&
+	 crush->buckets[-1-id]) {
```

Comment:
as a follow up, we could revise this part like
```c++
for (int index = 0; index < crush->max_buckets; index++) {
  if (crush->buckets[index] == nullptr) {
    return -index - 1;
  }
}

// enlarge the buckets array
// ...
```

Reply:
Done